x/core/types/data_request.go

```diff
@@ -14,6 +18,47 @@ func (i DataRequestIndex) DrID() string {
 	return string(i[24:])
 }
 
+// Strings returns the string-slice representation of the data request index.
+// It returns nil if the index is not valid.
+// [gas_price, height, dr_id]
+func (i DataRequestIndex) Strings() []string {
```

Comment:
does go let us do `[3]string` as the return type? Cause general rule of thumb is if we have a known size array its better to use that, than a dynamic one. For security, bug, and optimization reasons.